src/main/scala/nodes/learning/KernelGenerator.scala

```diff
@@ -0,0 +1,273 @@
+package nodes.learning
+
+import breeze.linalg._
+import breeze.math._
+import breeze.numerics._
+import org.apache.spark.rdd.RDD
+import org.apache.spark.SparkConf
+import org.apache.spark.SparkContext
+import org.apache.spark.SparkContext._
+import utils._
+
+
+// Trait to be implemented by Kernel functions
+trait KernelGenerator[T] {
```

Comment:
I think we need to store the transformation for it to be the same across train, test ? I agree as generally defined, a Kernel is K(x, y) = <value> i.e. it is defined on two data points. What we are doing here is fixing one of them, so we are defining K1(x) = K(x, trainRDD). Now you can replace trainRDD with random gaussian and you get RF. Thats my mental model fwiw